src/index.ts

```diff
@@ -270,7 +277,13 @@ export const handleClearKey = async (ctx: Context, _request?: Request) => {
 
 	for (let i = 0; i < keys.objects.length; i++) {
 		const key = keys.objects[i];
-		const keyUploadTime = new Date(key.uploaded);
+		const notBefore = key.customMetadata?.notBefore;
+		let keyUploadTime: Date;
+		if (notBefore) {
+			keyUploadTime = new Date(Number.parseInt(notBefore) * 1000);
```

Comment:
Shouldn't `keyUploadTime` be updated to have a new name? `keyStartTime` perhaps?
This line is a bit strange with the current naming. Could call it `keyNotBefore` since that is what is being calculated, thoughts?

Reply:
i did not have a good name, but `keyNotBefore` is a good suggestion